io.github.martchus.syncthingtray.yml

```diff
@@ -0,0 +1,147 @@
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.KGlobalSettings
+  - --talk-name=com.canonical.AppMenu.Registrar
+  - --talk-name=org.kde.kconfig.notify
+rename-appdata-file:
+  syncthingtray.appdata.xml
+rename-desktop-file:
+  syncthingtray.desktop
+rename-icon:
+  syncthingtray
+cleanup:
+  - /ForkAwesome
+  - /include
+  - /lib/cmake
+  - /lib/perl5
+  - /lib/pkgconfig
+  - /share/c++utilities
+  - /share/man
+  - /share/qtforkawesome
+  - /share/qtforkawesomeiconengine
+  - /share/qtquickforkawesome
+  - /share/qtutilities
+  - /share/syncthingconnector
+  - /share/syncthingmodel
+  - /share/syncthingwidgets
+  - '*.cmake'
+  - '*.la'
+  - '*.a'
+modules:
+  - name: syncthing
+    buildsystem: simple
+    build-options:
+      env:
+        GOBIN: /app/bin
+    build-commands:
+      - |
+        . /usr/lib/sdk/golang/enable.sh
+        export GOPATH=$PWD/go
+        go version
+        # go mod vendor
+        #go build -mod=vendor
+        #go mod download -json > download.json
+        env GOFLAGS=-mod=vendor go run build.go -no-upgrade build
+      - install -Dm755 -t /app/bin syncthing
+    sources:
+      # Use git will raise `inconsistent vendoring` error, so use archive here.
+      - type: archive
+        url: https://github.com/syncthing/syncthing/releases/download/v1.27.7/syncthing-source-v1.27.7.tar.gz
+        sha256: 2572a2fbb425be2998ff045ea08055710182ced5b3673b96107ccaa56b34c5b5
+  - name: boost
+    buildsystem: simple
+    build-commands:
+      - ./bootstrap.sh --prefix=/app
+      - ./b2 --with-filesystem -j $FLATPAK_BUILDER_N_JOBS link=static install
+    sources:
+      - type: archive
+        url: https://boostorg.jfrog.io/artifactory/main/release/1.85.0/source/boost_1_85_0.tar.gz
+        sha256: be0d91732d5b0cc6fbb275c7939974457e79b54d6f07ce2e3dfdd68bef883b0b
+  - name: cpp-utilities
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    sources:
+      - type: git
+        url: https://github.com/Martchus/cpp-utilities.git
+        tag: v5.24.8
+      - type: patch
+        path: patches/0001-Specify-the-developer-ID-in-AppStream-meta-data.patch
+  - name: qtutilities
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE:STRING=Release
+      - -DQT_PACKAGE_PREFIX:STRING='Qt6'
+      - -DBUILTIN_TRANSLATIONS:BOOL=ON
+    sources:
+      - type: git
+        url: https://github.com/Martchus/qtutilities.git
+        tag: v6.14.0
+  - name: ForkAwesome
+    buildsystem: simple
+    build-commands:
+      - install -D ./fonts/forkawesome-webfont.woff2 /app/ForkAwesome/forkawesome-webfont.woff2
+      - install -D ./src/icons/icons.yml /app/ForkAwesome/icons.yml
+    sources:
+      - type: archive
+        url: https://github.com/ForkAwesome/Fork-Awesome/archive/refs/tags/1.2.0.tar.gz
+        sha256: 23fba5f191f204e0414c547bf4c9b10fd7ca42c151260e8f64698449a75fbdb3
+  - name: perl-libs
```

Comment:
are these used only at build time?

Reply:
Perl is only used at build time.

Reply:
then this module should have a local `'*'` cleanup rule.

Reply:
> then this module should have a local `'*'` cleanup rule.

OK. But I am a bit confused: The flatpak-builder seems only accept changes in `/app` directory, but the build directory for module sources is not the `/app`. So in such case the local '*' cleanup rule seems doing nothing?

Reply:
It will  cleanup all contents of that module in the final stage from `/app`.

Reply:
it work the same way as the top level cleanup except it is only on what the module installed. It's explained in the documentation.

Since it is only a build time dependency then it make sense to remove it all at the end.

Reply:
Wow! That is a genius design!